Expands Valhalla With A Vehicle Area, Adds Assault Vehicle Category to Valhalla Req Vendor

Wisemonster opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Expands Valhalla with an area for testing vehicles (currently only the tank), and adds a vehicle ammo category to the valhalla req vendor, currently only contains the tank ammo. Requires the tank elevators/vendor to be changed to allow multiple vehicles across z-levels before it can be added to valhalla.
![tgmc  _maps_map_files_generic_Admin_Level dmm  - StrongDMM 4_20_2024 7_55_50 PM](https://github.com/tgstation/TerraGov-Marine-Corps/assets/87689371/9b4347fb-4b5d-4fbb-880e-fb1111f51590)



<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Prepares Valhalla for testing Assault vehicles whenever they are fully merged.
<!-- Argue for the merits of your changes and how they benefit the game, especially if they are controversial and/or far reaching. If you can't actually explain WHY what you are doing will improve the game, then it probably isn't good for the game in the first place. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
add: Expanded west side of Valhalla with an assault vehicle area
add: Added an Assault Vehicle category to the Valhalla requisition vendor; added tank ammo to said category
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Wisemonster commented:
Note: the picture was before I added a bunch of cosmetic oil splatters around the assault vehicle area.

Wisemonster commented:
Ok, it looks like I might have to replace the lattices next to the rwalls on the z-level, cause otherwise it's runtiming

Wisemonster commented:
Ok, turns out what the linter means by neighbor is something on the same tile, so I just gotta fix some lattices on the same tiles as the walls

TiviPlus commented on code/game/objects/machinery/vending/marine_vending.dm:
```diff
@@ -1505,6 +1505,7 @@
 			/obj/structure/largecrate/supply/weapons/heavy_flakgun = -1,
 			/obj/item/ammo_magazine/heavy_isg/he = -1,
 			/obj/item/ammo_magazine/heavy_isg/sabot = -1,
+
 		),
```
```suggestion
		),
```

TiviPlus commented on code/game/objects/machinery/vending/marine_vending.dm:
```diff
@@ -1537,6 +1538,11 @@
 			/obj/item/implanter/chem/blood = -1,
 			/obj/item/implanter/blade = -1,
 		),
+		"Assualt Vehicle" = list(
```
```suggestion
		"Assault Vehicle" = list(
```

TiviPlus reviewed (changes requested):
Tank vendor currently only supports one vend of assault/non assault and this is shared globally

Wisemonster commented:
Alright. I could place a tank in the area as a sort-of quick fix, or I could wait until multiple assault vehicles are spawnable then update the area with the tank vendor.

Wisemonster commented:
> Tank vendor currently only supports one vend of assault/non assault and this is shared globally

To clarify, you mean the tank vendor connected to the elevator, not the tank parts vendor, right?

TiviPlus commented:
> > Tank vendor currently only supports one vend of assault/non assault and this is shared globally
> 
> To clarify, you mean the tank vendor connected to the elevator, not the tank parts vendor, right?

Uhh wdym they're the same thing, but theres a globally tracked "this vehicle has been vended"

TiviPlus commented:
You can do the place a few tanks workaround as a lazy fix and people blowing them up can cope but it should be changed eventualy

Wisemonster commented:
Alright, updated pr to add a couple of tanks to the area, along with the parts for the tanks' primary and secondary slots. Removed the fabricator cause it's not going to do anything except maybe cause problems for living assault/transport crewmen till multiple vehicle elevators can exist.

TiviPlus commented:
@RipGrayson

RipGrayson reviewed (changes requested):
Please move this carp somewhere else, two tiles away from any walls/windows or Valhalla's interior
![carp7](https://github.com/tgstation/TerraGov-Marine-Corps/assets/49290523/5b18d9d3-18bd-45a3-9ac7-ce30ca9fbc94)
You have space tiles here
![carp8](https://github.com/tgstation/TerraGov-Marine-Corps/assets/49290523/4ed6d949-bfcc-4d6f-bbcc-f8fa5958b606)

Wisemonster commented:
Alright, moved the carp and replaced the space tiles
